services/workflows-service/src/alert/alert.controller.external.ts

```diff
@@ -6,42 +6,107 @@ import { PrismaService } from '@/prisma/prisma.service';
 import { CurrentProject } from '@/common/decorators/current-project.decorator';
 import { AppLoggerService } from '@/common/app-logger/app-logger.service';
 import { CreateAlertDefinitionDto } from './dtos/create-alert-definition.dto';
-import { AlertDefinition, Alert } from '@prisma/client';
-import { type TProjectId } from '@/types';
+import { AlertDefinition, Alert, Prisma } from '@prisma/client';
+import type { TProjectId, TProjectIds } from '@/types';
 import * as errors from '../errors';
 import { ProjectIds } from '@/common/decorators/project-ids.decorator';
 import { ZodValidationPipe } from '@/common/pipes/zod.pipe';
-import { Body, Controller, Get, Post, Query, UsePipes } from '@nestjs/common';
+import * as common from '@nestjs/common';
+import {
+  AlertAssigneeUniqueDto,
+  AlertsIdsByProjectDto,
+  BulkAssignAlertsResponse,
+} from './dtos/assign-alert.dto';
+import { BulkStatus, TBulkAssignAlertsResponse } from './types';
+import { ProjectAssigneeGuard } from '@/alert/guards/project-assignee.guard';
 
 @swagger.ApiBearerAuth()
 @swagger.ApiTags('Alerts')
-@Controller('external/alerts')
+@common.Controller('external/alerts')
 export class AlertControllerExternal {
   constructor(
     protected readonly service: AlertService,
     protected readonly prisma: PrismaService,
     protected readonly logger: AppLoggerService,
   ) {}
-  @Post()
+  @common.Post()
   @swagger.ApiCreatedResponse({
     type: 'string',
   })
   @UseCustomerAuthGuard()
   @swagger.ApiForbiddenResponse()
   async create(
-    @Body() createAlertDto: CreateAlertDefinitionDto,
+    @common.Body() createAlertDto: CreateAlertDefinitionDto,
     @CurrentProject() currentProjectId: TProjectId,
   ): Promise<AlertDefinition> {
     // Assuming create method in AlertService accepts CreateAlertDto and returns AlertDefinition
     return await this.service.create(createAlertDto, currentProjectId);
   }
 
-  @Get('/')
+  @common.Get('/')
   @swagger.ApiOkResponse({ type: Array<Object> }) // TODO: Set type
   @swagger.ApiNotFoundResponse({ type: errors.NotFoundException })
   @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
-  @UsePipes(new ZodValidationPipe(FindAlertsSchema, 'query'))
-  async getAll(@Query() findAlertsDto: FindAlertsDto, @ProjectIds() projectIds: TProjectId[]) {
+  @common.UsePipes(new ZodValidationPipe(FindAlertsSchema, 'query'))
+  async getAll(
```

Comment:
Please follow existing naming conventions, we're open to changing them but in an organized manner. Can be `listAlerts` or similar too.
```suggestion
  async list(
```